doc.go

```diff
@@ -177,7 +179,54 @@ cmd/eskip command package, and/or enter in the command line:
 Extending Skipper
 
 Skipper doesn't use dynamically loaded plugins, but it can be used as a
-library and extended with custom filters and/or custom data sources.
+library and extended with custom filters, predicates and data sources.
+
+
+Custom Predicates
+
+To create a custom predicate, the PredicateSpec of the routing package
+needs to be implemented. The PredicateSpec is used to create the actual
+predicate with concrete arguments, during the route definitions are
+processed by the routing package.
```

Comment:
yeah... (shyface)

one needs to provide the 'factory' object, the spec, and its create method should return the actual predicate object, and it is called when the routing tree is constructed. can we hammer it out together?